### PR TITLE
xtest: add 6016 to test concurency storage

### PR DESCRIPTION
**Pull request by jforissier**
This testcase against to RPMB secure storage.
Create multiple threads to do secure storage, before the
modify that move lock to rpmb fs layer, it will fail.

Change-Id: I4138f8f7332a0038378b86ed7dc111cf1c6ad86b
Signed-off-by: Lijianhui (Airbak) <airbak.li@hisilicon.com>
[Restricted test to RPMB only, removed begin/end subcase]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

File: host/xtest/xtest_6000.c [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include <string.h>
 #include <stdio.h>
 
+#ifdef USER_SPACE
```
Comment (jenswi-linaro) [inline host/xtest/xtest_6000.c]: I think we can assume always user space by now, please remove the `#ifdef`

File: host/xtest/xtest_6000.c [hunk 2]
```diff
@@ -1459,6 +1463,126 @@ static void xtest_tee_test_6015_single(ADBG_Case_t *c, uint32_t storage_id)
...
+{
+	struct test_6016_thread_arg *a = arg;
+	TEEC_Session sess;
+	uint32_t obj;
+	uint8_t out[10] = { 0 };
+	uint32_t count;
+	uint32_t orig;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+		xtest_teec_open_session(&sess, &storage_ta_uuid, NULL, &orig)))
```
Comment (jenswi-linaro) [inline host/xtest/xtest_6000.c]: The concurrency would be improved if the sessions were created before the threads.

**Comment by jforissier** [comment]
@Airbak FYI

File: host/xtest/xtest_6000.c (hunk 1 above)
Comment (jforissier) [inline host/xtest/xtest_6000.c]: OK

File: host/xtest/xtest_6000.c (hunk 2 above)
Comment (jforissier) [inline host/xtest/xtest_6000.c]: OK

**Comment by jforissier** [comment]
Updated

**Comment by airbak** [comment]
@jenswi-linaro thanks for your review.
@jforissier thanks for your update,  it looks good to me, sorry to trouble you!

File: host/xtest/xtest_6000.c [hunk 3]
```diff
@@ -1459,6 +1460,130 @@ static void xtest_tee_test_6015_single(ADBG_Case_t *c, uint32_t storage_id)
...
+	size_t n;
+	size_t m;
+
+	memset(arg, 0, sizeof(arg));
+
+	for (m = 0; m < num_threads; m++)
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+			xtest_teec_open_session(&arg[m].session,
+				&storage_ta_uuid, NULL, &orig)))
+			goto out;
```
Comment (jenswi-linaro) [inline host/xtest/xtest_6000.c]: Since thr[] doesn't hold valid thread ids we'll have undefined behavior with `pthread_join()` below.

Comment (jforissier) [inline host/xtest/xtest_6000.c]: Fixed.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**